client/src/three/GameRenderer.ts

```diff
@@ -174,6 +157,28 @@ export default class GameRenderer {
     this.animate();
   }
 
+  private setupListeners() {
+    window.addEventListener("urlChanged", this.handleURLChange);
+    window.addEventListener("resize", this.onWindowResize.bind(this));
+  }
+
+  private handleURLChange = () => {
+    const url = new URL(window.location.href);
+
+    const scene = url.pathname.split("/").pop();
+
+    if (scene === this.sceneManager.getCurrentScene() && this.sceneManager.getCurrentScene() === SceneName.WorldMap) {
```

Comment:
wouldn't it be easier to add the moveCameraFromScene inside the switch scene method ? is there any place where we switch scene without moving the camera ?